Libraries/LLM/Evaluate.swift

```diff
@@ -58,8 +58,8 @@ public struct TokenIterator: Sequence, IteratorProtocol {
 
     mutating public func next() -> MLXArray? {
         var logits: MLXArray
-        (logits, cache) = model(expandedDimensions(y, axis: 0), cache: cache.isEmpty ? nil : cache)
-        y = sample(logits: logits[-1, axis: 1], temp: temp, topP: topP)
+        (logits, cache) = model(y[.newAxis], cache: cache.isEmpty ? nil : cache)
+        y = sample(logits: logits[0..., -1, 0...], temp: temp, topP: topP)
```

Comment:
The second line matches:

- https://github.com/ml-explore/mlx-examples/blob/main/llms/mlx_lm/utils.py#L173

```python
        logits = logits[:, -1, :]
```